### PR TITLE
LPS-184042 Refactor the UAD portlet to use string keys rather than class keys

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-184042](https://issues.liferay.com/browse/LPS-184042).

@carolmariaabb @gabrielwas @marcelabc Here is my attempt. I have tested nothing so far, but if you are welcome to begin building on top of this. If you do, please let me know if you encounter any issues. Thanks!

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:help

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
Thanks a lot @drewbrokke 🚀

**Comment by drewbrokke** [comment]
ci:test:um

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:um

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:um

**Comment by drewbrokke** [comment]
@patriciajeanperez do the `um` suite failures look related to UAD?

File: modules/apps/user-associated-data/user-associated-data-web/src/main/java/com/liferay/user/associated/data/web/internal/portlet/action/BaseUADMVCActionCommand.java
Comment (pei-jung) [inline modules/apps/user-associated-data/user-associated-data-web/src/main/java/com/liferay/user/associated/data/web/internal/portlet/action/BaseUADMVCActionCommand.java:1]: The `entityType` in `_getUADRegistryKey ` should be escaped too.

**Review by pei-jung — CHANGES_REQUESTED** [review]
Just one small spot that I think you missed. Everything else looks good! 👍

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**